Add Blessed Wine prayer automation plugin

TheHaliax opened the pull request:
Introduces the Blessed Wine plugin for automating prayer training using blessed wines. Includes configuration, overlay, main plugin logic, state management, and the core script to handle banking, walking, blessing, and prayer restoration cycles.

TheHaliax committed:
Add Blessed Wine prayer automation plugin

Introduces the Blessed Wine plugin for automating prayer training using blessed wines. Includes configuration, overlay, main plugin logic, state management, and the core script to handle banking, walking, blessing, and prayer restoration cycles.

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/hal/blessedwine/BlessedWinePlugin.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    @Subscribe
+    public void onGameTick(GameTick tick) {
```
It would be best to remove events that are unused.

TheHaliax replied:
my bad i had planned on doing more but realized most of it wasn't required and forgot to delete the extras

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/hal/blessedwine/BlessedWineScript.java:
```diff
@@ -0,0 +1,256 @@
+@Slf4j
+public class BlessedWineScript extends Script {
+    @Inject
+    private ConfigManager configManager;
```
Instead of injecting your own instance of the clientThread or config manager, lets use Microbot.getConfigManager() or Microbot.getClientThread().

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/hal/blessedwine/BlessedWineScript.java:
```diff
@@ -0,0 +1,256 @@
+                case INITIALIZING:
+                    BlessedWinePlugin.status = "Initializing run...";
+                    initialize();
+                    if (!super.run()) return;
```
This is not proper usage of the super.run() method. This should be placed as the first line within your future.

With how you currently have it, this will only run when in the initialize state.

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/hal/blessedwine/BlessedWineScript.java:
```diff
@@ -0,0 +1,256 @@
+                    BlessedWinePlugin.status = "Teleporting back to Cam Torum Bank...";
+                    if (Rs2Player.getWorldLocation().getPlane() != 2) {
+                        for (int id : CALCIFIED_MOTH) {
+                            if (Rs2Inventory.hasItem(id)) {
```
Inside of any inner loops, we should be first checking `if (!isRunning()) break;`, this ensures that when the scheduled future is cancelled that we break out of the for loop, instead of continuing.

TheHaliax committed:
Refactor BlessedWineScript to remove unused dependencies

Removed unused imports, dependency injections, and constructor parameters from BlessedWineScript. Added isRunning() checks in loops to improve script control and prevent unnecessary actions when the script is stopped.

TheHaliax committed:
Remove unused onGameTick event handler

Deleted the onGameTick method from BlessedWinePlugin as it was unused and contained only a placeholder comment.
